docs/Projects/TAD/ADRs/0004-software-stack.md

```diff
@@ -0,0 +1,81 @@
+# ADR-0004 Software Stack for TAD
+
+## Context
+
+For building our own TAD solution, we need to choose a software stack. During our earlier
+POCs and market research, we gathered insight and information on technologies to use
+and which not to use.
+
+During further discussions and brainstorm sessions, a software stack was chosen
+that accommodates our needs best.
+
+While more fine grained [requirements](../Existing-Tools/Comparison/requirements) are listed elsewhere,
+some key requirements are:
+
+* the tool is runnable on local dev machines
+* the tool is runnable as user as a local Docker solution
+* the tool is runnable on a cloud platform based on Kubernetes
+* the tool supports multiple organizations, teams and projects
+* the tool supports oath / keycloak
+
+## Assumptions
+
+We stick to one programming language to avoid unneeded complexity. As most AI related
+tooling is written in Python, this is the logical choice for our development as well.
+
+There is no need for a complex or separate web GUI framework. Also to avoid more components and complexity,
+it is preferred to bundle backend and frontend in one solution.
+
+As a government organization, we need to adhere to the digital inclusion standards, like
+[Drempelvrij](https://www.drempelvrij.nl/).
+
+We do not forsee high load or many static files, therefor a separate webserver like
+nginx, apache is not required. Static files will be served directly by the backend.
+
+## Decision
+
+### Programming language
+
+We will use [Python](https://www.python.org/) v3.11.X as programming language
+and [Poetry](https://python-poetry.org/) for dependency management.
+
+### Backend
+
+The Python backend will use the following key dependencies:
+
+* [Granian](https://github.com/emmett-framework/granian) as HTTP server
+* [Pydantic](https://docs.pydantic.dev/latest/) for data validation
+* TODO: decide on [FastAPI](https://fastapi.tiangolo.com/) or
+  [Flask](https://flask.palletsprojects.com/en/3.0.x/)
+
+### Frontend
+
+We will use an integrated GUI solution, based on [HTMX](https://htmx.org/) and
+TODO: decide what template framework to use, like [Jinja](https://jinja.palletsprojects.com/en/3.1.x/).
+This combination is expected to work intuitive and fast.
+
+TODO: decide which 'frontend' to use, like [DaisyUI](https://daisyui.com/) or more
+custom [NL Design System](https://github.com/nl-design-system/rijkshuisstijl-community).
+
+TODO: if we use HTMX, validating web responses may be more difficult, as we get full HTML
+structures instead of 'simple' JSON objects
+
+### Testing
+
+We will use [pytest](https://docs.pytest.org/en/8.0.x/) for unit-testing
+and [Playwright](https://playwright.dev/)
+for front-end and integration tests.
+TODO: decide on [VCRPY](https://vcrpy.readthedocs.io/en/latest/) or [Mock Server](https://pypi.org/project/mock-server/).
+
+### Database
+
+TODO: decide on [SQL Alchemy](https://www.sqlalchemy.org/) or [SQLite](https://www.sqlite.org/) or other?
```

Comment:
I don't think this is the choice that we need to make, it would be SQLAlchemy **and** SQLite. Or SQLAlchemy and Postgress. 

A choice we may still want to make is whether we'd like relational databases vs document stores

Reply:
You're right, my mistake.